lib/common/bits.h

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_BITS_H
+#define ZSTD_BITS_H
+
+#include "mem.h"
+
+MEM_STATIC unsigned ZSTD_highbit32_fallback(U32 val) {
+    static const U32 DeBruijnClz[32] = {  0,  9,  1, 10, 13, 21,  2, 29,
+                                         11, 14, 16, 18, 22, 25,  3, 30,
+                                          8, 12, 20, 28, 15, 17, 24,  7,
+                                         19, 27, 23,  6, 26,  5,  4, 31 };
+    val |= val >> 1;
```

Comment:
minor : 
(and mostly just an argument in favor of defensive programming).
I note you haven't added an `assert(val != 0)` here,
probably because it's already in `ZSTD_highbit32()` below.

But the point of adding `assert()` is not just that it works, today,
but that it's more difficult, in the future, to make modifications or bad usage of this function.

In this case, if you don't `assert(val != 0)`,
it implicitly means that this function could work with `val == 0`, 
irrespective of the fact that it's never called in this configuration ... yet.
And by the way, it superficially does : if `val == 0`, this function cleanly returns `0`, so it could seem that everything works as intended. There will be consequences in later parts of the code, and it might take a while to get back to the source of the issue.

Recommendation : even though the function is never called with `val == 0`, it should `assert()` it inside its own body, as the function could be re-employed in some different context in the future.
Also, you may want to add a code comment to make this condition clear, or explain why it exists (which is something the `assert()` alone cannot do).

Reply:
Agreed, I added this assert on my own fork -- will force-push here once I get that update to pass CI.